p2panda-sync/src/log_sync.rs

```diff
@@ -260,20 +261,16 @@ async fn remote_needs<T, E>(
 where
     E: Clone + Serialize,
 {
-    let mut log = store
-        .get_log(public_key, log_id)
+    let log = store
+        .get_raw_log(public_key, log_id)
         .await
-        .map_err(|e| SyncError::Protocol(e.to_string()))?;
+        .map_err(|err| SyncError::Custom(format!("could not retrieve log from store: {err}")))?;
```

Comment:
This is strictly speaking no `Protocol` error but an critical storage failure. `Custom` feels too generic though as we know quite well what the category of this error is. Should I try to wrap it in an `Io` type?

Reply:
I'd go with the `Io` wrapper here.

Reply:
Hmn, I looked a little bit into Input/Output APIs and I'm not sure databases really fit in there. I'll gather some thoughts, this feels like a separate PR, will write about it there!

Reply:
See https://github.com/p2panda/p2panda/pull/575